backend/src/controllers/loginController.ts

```diff
@@ -0,0 +1,63 @@
+import { Request, Response } from "express";
+import jwt from "jsonwebtoken";
+import bcrypt from "bcrypt";
+import { Farmer } from "../models/farmer";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+interface LoginRequestBody {
+  email?: string;
+  password?: string;
+}
+
+const loginController = async (
+  req: Request<{}, {}, LoginRequestBody>,
+  res: Response
+) => {
+  const { email, password } = req.body;
+
+  // Validate credentials
+  if (!email || !password) {
+    return res.status(400).json({ message: "Email and password are required" });
+  }
+
+  try {
+    // Check if farmer exists
+    const farmer = await Farmer.findOne({ email }).select("+password");
```

Comment:
What does +password do?

Reply:
Because I have set password selecte false, which means if we want to read that information, we need to add it explicitly in the query. This is for security purposes only.

const farmerSchema = new Schema<IFarmer>({
  email: { type: String, required: true, unique: true },
  password: { type: String, required: true, select: false },
});